### PR TITLE
LPS-129548, LPS-129217, LPS-129214 | Master

### PR DESCRIPTION
**Pull request by liuxingye**
https://issues.liferay.com/browse/LPS-129548
hi @austinchiang , I've added one more case ^ for questions to validate if the guest user cannot ask quesion.

https://issues.liferay.com/browse/LPS-129217
https://issues.liferay.com/browse/LPS-129214

### CODE REVIEW

**Comment by liuxingye** [comment]
ci:test:sf

**Comment by austinchiang** [comment]
**Reviewing...** :hourglass_flowing_sand: :sunglasses: :headphones:

File: portal-web/test/functional/com/liferay/portalweb/macros/ItemSelector.macro [hunk 1]
```diff
@@ -758,6 +758,58 @@ definition {
...
+	macro viewFilteredResultsViaAssetLists {
+		var key_assetType = "${assetType}";
+
+		if (isSet(assetListName)) {
+			Click(
+				key_title = "${assetListName}",
+				locator1 = "AssetLists#ASSET_LIST_ENTRY_TITLE");
+		}
+
+		Pause(locator1 = "3000");
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/macros/ItemSelector.macro:770]: Hey @liuxingye I noticed in the last couple of pulls that you've been using a lot of hard pauses. Can we try to use wait functions (like assertElementPresent) instead?

File: modules/apps/item-selector/item-selector-test/src/testFunctional/tests/ItemSelector.testcase [hunk 2]
```diff
@@ -244,6 +244,66 @@ definition {
 			viewNoLocationInfo = "Site Name");
 	}
 
+	@description = "This test ensures that a folder can be filtered from the current asset library."
+	@priority = "5"
+	test FilterFolderFromCurrentAL {
```
Comment (austinchiang) [inline modules/apps/item-selector/item-selector-test/src/testFunctional/tests/ItemSelector.testcase:249]: We need to refactor our naming at some point. We have AL, AssetLibrary, AssetLibraries, Depot as names in our Poshi tests. Let's try to use one name for all of them

Comment (liuxingye) [reply modules/apps/item-selector/item-selector-test/src/testFunctional/tests/ItemSelector.testcase:249]: sure, I can file a test improvement ticket to refactor the name later.

File: portal-web/test/functional/com/liferay/portalweb/macros/ItemSelector.macro (hunk 1 above)
Comment (liuxingye) [reply portal-web/test/functional/com/liferay/portalweb/macros/ItemSelector.macro:770]: I refer to AssetListsAdmin.selectAsset to write the new macro, it also works without pause locally, but to be more stable, I added a pause finally.

Comment (liuxingye) [reply portal-web/test/functional/com/liferay/portalweb/macros/ItemSelector.macro:770]: I checked my commits from last Tuesday, I used pause for 2 times. one is here, another is [ViewMentionViaBlogsEntryCommentAsGuest](https://github.com/austinchiang/liferay-portal/pull/986/files), use pause there to give enough time loading for blog things.

File: modules/apps/item-selector/item-selector-test/src/testFunctional/tests/ItemSelector.testcase (hunk 2 above)
Comment (liuxingye) [reply modules/apps/item-selector/item-selector-test/src/testFunctional/tests/ItemSelector.testcase:249]: see https://issues.liferay.com/browse/LRQA-66061

Comment (austinchiang) [reply modules/apps/item-selector/item-selector-test/src/testFunctional/tests/ItemSelector.testcase:249]: Thanks! Yeah let's use Depot whenever possible

**Comment by liuxingye** [comment]
Will send a new pull with some test improvements.

**Comment by liuxingye** [comment]
see https://github.com/austinchiang/liferay-portal/pull/995

**Pull request closed without merging**